src/Functions/FunctionsHashing.h

```diff
@@ -1688,6 +1688,25 @@ struct ImplWyHash64
     static constexpr bool use_int_hash_for_pods = false;
 };
 
+struct ImplEntropyLearnedHash
+{
+    static constexpr auto name = "entropyLearnedHash";
+    // TODO: made also a function, which would set positions by training the method on some dataset
+    static std::vector<size_t> positions;
+    using ReturnType = UInt64;
+
+    static auto combineHashes(UInt64 h1, UInt64 h2) { return CityHash_v1_0_2::Hash128to64(CityHash_v1_0_2::uint128(h1, h2)); }
+    static auto apply(const char * s, const size_t len) {
+        // TODO: why undefined symbols of using ctor of Key, positions, GetBytes? 
```

Comment:
I think that is because CMake doesn't glob source files in `contrib/`. (more specifically, src/Functions/CMakeLists.txt contains `include("${ClickHouse_SOURCE_DIR}/cmake/dbms_glob_sources.cmake")
add_headers_and_sources(clickhouse_functions .)` but there is no equivalent code in contrib/CMakeLists.txt.). I think the problem will be resolved once you move all code to `src/Functions/` (see my other comment).

Reply:
Yes, moving the code resolved the problem, thanks